gc/default/default.c

```diff
@@ -6261,6 +6274,15 @@ rb_gc_impl_ractor_cache_free(void *objspace_ptr, void *cache)
     rb_objspace_t *objspace = objspace_ptr;
 
     objspace->live_ractor_cache_count--;
+    rb_ractor_newobj_cache_t *newobj_cache = (rb_ractor_newobj_cache_t *)cache;
+
+    for (size_t heap_idx = 0; heap_idx < HEAP_COUNT; heap_idx++) {
+        rb_heap_t *heap = &heaps[heap_idx];
+        rb_ractor_newobj_heap_cache_t *heap_cache = &newobj_cache->heap_caches[heap_idx];
+        // TODO: Do we need atomics here? We're always inside VM_LOCK.
```

Comment:
I think we probably do need the atomic. We're holding the VM lock, and we should assume also we have exclusive access to `newobj_cache` (because we're about to delete it 😅), but other threads not inside VM_LOCK could still be incrementing `heap->total_allocated_objects`